src/transformers/trainer.py

```diff
@@ -887,21 +907,70 @@ def _objective(trial):
             best_trial = study.best_trial
             best_run = BestRun(str(best_trial.number), best_trial.value, best_trial.params)
         elif self.hp_search_backend == HPSearchBackend.RAY:
-            # The TensorBoard writer does not pickle so we have to remove it (if it exists) while doing the ray hp
-            # search.
+            # The model and TensorBoard writer do not pickle so we have to remove them (if they exists)
+            # while doing the ray hp search.
             _tb_writer = self.tb_writer
             self.tb_writer = None
+            _model = self.model
```

Comment:
There is no need to save the old model IMO. Being erased by a new instance of model_init is expected behavior when training.